Cabal/src/Distribution/Simple/Glob.hs

```diff
@@ -431,17 +430,37 @@ runDirFileGlob verbosity mspec rawRoot pat = do
       concat <$> traverse (\subdir -> go globPath (dir </> subdir)) subdirs
     go GlobDirTrailing dir = return [GlobMatch dir]
 
-  directoryExists <- doesDirectoryExist (root </> joinedPrefix)
```

Comment:
This line is duplicated inside the `case` now. Why moving it at all?

Reply:
It's not exactly the same - one does `doesDirectoryExist (root </> joinedPrefix </> filename)` and the other does `doesDirectoryExist (root </> joinedPrefix)`. 

The `Left filename` case is there so we don't report that a *file* exists when it's actually a directory - `doesFileExist `/home/` returns `True` and would give an incorrect result - ie `GlobMatchesDirectory`.

The `Right pattern` ensures that we give a `GlobMissingDirectory` if the path doesn't exist.

Reply:
Oh, my bad!